File: csharp/Multiton/Camera.cs
public class Camera
{
    static Dictionary<string, Camera> Cameras = new Dictionary<string, Camera>();
    static object _lock = new object();

    private Camera()
    {
        
    }

    public static Camera GetCamera(string brand)
    {
        lock (_lock)
        {
            if (!Cameras.ContainsKey(brand))
            {
                Cameras.Add(brand, new Camera());
            }
        }

        return Cameras[brand];

    }
}

File: java/multiton/src/main/Camera.java
package main;

import java.util.HashMap;

public class Camera {

    private static volatile HashMap<String, Camera> cameras = new HashMap<>();

    private Camera() {}

    public static Camera getCamera(String brand)
    {
        synchronized (Camera.class)
        {
            if (!cameras.containsKey(brand))
            {
                cameras.put(brand, new Camera());
            }
        }

        return cameras.get(brand);
    }

    @Override
    protected Object clone() throws CloneNotSupportedException {
        throw new CloneNotSupportedException();
    }
}
